src/object_store.rs

```diff
@@ -281,34 +281,26 @@ impl Object {
 impl io::Read for Object {
     fn read(&mut self, buffer: &mut [u8]) -> io::Result<usize> {
         if !self.remaining_bytes.is_empty() {
-            let len = cmp::max(buffer.len(), self.remaining_bytes.len());
-            buffer.copy_from_slice(&self.remaining_bytes[..len]);
-
-            if self.remaining_bytes.len() > len {
-                self.remaining_bytes = self.remaining_bytes[len..].to_vec();
-            }
-
+            let len = cmp::min(buffer.len(), self.remaining_bytes.len());
+            buffer[..len].copy_from_slice(&self.remaining_bytes[..len]);
+            self.remaining_bytes = self.remaining_bytes[len..].to_vec();
             return Ok(len);
         }
 
-        if self.skip_next {
+        if !self.skip_next {
             self.skip_next = false;
```

Comment:
If line 291 executes, `self.skip_next` is already false. Either there's a logic bug, or line 291 can be removed.

(What is the purpose of skip_next)?

Reply:
Agree, line 291 can be removed.

Bug was having condition on true, instead of false. Because of that, it never actually executed.

The purpose of next is to:

subsribe to object_store subject and get messages containing data. Each loop check if there are more pending messages for the subject. If yes - set skip_next to false so next read iteration fetches next messages. If no and we're done - set it to true. Than each `read` iteration will only move data to buffer from remaining bytes.

I guess variable name can be changed to something describing actual behaviour.

Reply:
@stevelr does it look clear to you know?